tests/test_engine.py

```diff
@@ -1034,3 +1038,167 @@ def my_flow():
         task_runs = await orion_client.read_task_runs()
 
         assert sorted([int(run.dynamic_key) for run in task_runs]) == [0, 0, 1, 1]
+
+
+class TestLinkStateToResult:
+
+    state = State(id=uuid4(), type=StateType.COMPLETED)
+
+    class MockFlowRunContext:
+        def __init__(self):
+            self.task_run_results = {}
+
+    class RandomTestClass:
+        pass
+
+    class PydanticTestClass(BaseModel):
+        untrackable_num: int
+        list_of_ints: List[int]
+
+    @pytest.mark.parametrize(
+        "test_input", [True, False, -5, 0, 1, 256, ..., None, NotImplemented]
+    )
+    def test_link_state_to_result_with_untrackables(self, monkeypatch, test_input):
+        ctx = self.MockFlowRunContext()
+
+        def get():
+            return ctx
+
+        monkeypatch.setattr("prefect.engine.FlowRunContext.get", get)
```

Comment:
This feels like a hack. I'd try to avoid mocks unless you're making assertions about calls. Let's introduce a `flow_run_context` fixture that builds and enters a fake flow run context for the `flow_run` fixture.